src/main/scala/se/nimsa/dcm4che/streams/DicomParsing.scala

```diff
@@ -55,6 +72,82 @@ trait DicomParsing {
     }
   }
 
+  // parse dicom UID data element from buffer
+  def parseUIDDataElement(data: ByteString, explicitVR: Boolean, assumeBigEndian: Boolean): DataElement = {
+    val maybeHeader = if (explicitVR) {
+      readHeaderExplicitVR(data, assumeBigEndian)
+    } else {
+      readHeaderImplicitVR(data)
+    }
+
+    if (maybeHeader.isEmpty) {
+      throw new DicomStreamException("Could not parse DICOM data element from stream.")
+    }
+
+    val (tag, vr, headerLength, length) = maybeHeader.get
+    val value = data.drop(headerLength).take(length)
+    DataElement(tag, vr, length, valueWithoutPadding(value))
+  }
+
+
+  def valueWithoutPadding(value: ByteString) =
```

Comment:
Hmm, only used in 'parseUIDDataElement', so we know the characters used in the value are "0"-"9", "." of Default Character Repertoire (according to the Standard). So if it's 00H it must be padding. Maybe should nest this method in  'parseUIDDataElement'?